### ISSUE TITLE
BP: POST service request

### ISSUE THREAD

**Issue by harper-wavra** [USGS-WiM/StreamStats#1601]
Once all the user information is entered (from #1460), we need to:

- [x] Make a post request to https://streamstats.usgs.gov/batchprocessor/batch, with those inputs. The payload for that POST request needs to be a formdata object with a region, basinCharacteristics, flowStatistics, IDField, geometeryFile, & email.
    - This is an example for ND when I select Peakflow statistics:
```
region: ND
basinCharacteristics: DRNAREA,COMPRAT,RUGGED
flowStatistics: 2
IDField: localID
geometryFile: ND.zip
email: hwavra@usgs.gov
```
- [x] When that request completes, we should show a message saying it either completed successfully, or if there were any errors. 
- [x] If the user attempts to submit a shapefile with more than 250 points, the services will return a failure message ("Batch must contain fewer than 250 points."). When that happens, show a toast message and a yellow warning box on the modal with this message: ~"Batch must contain fewer than 250 points. Please resubmit the batch if you would like only the first 250 points to be processed."~ "Batch contains more than 250 points. Only the first 250 points will be processed. Please resubmit the batch if you would like only the first 250 points to be processed." Then, if the user submits the batch again, add the following parameter to the post request: `moreThan250Points: true`. This will allow the user to submit the batch.
     - We can use this .zip file for testing: https://www.sciencebase.gov/catalog/item/577445bee4b07657d1a991b6 (Local ID can be `SOURCE_FEA`)

Helpful Hints (maybe):
- Here are a couple upload files across different states you can use while testing.
    - [pointsND.zip](https://github.com/USGS-WiM/StreamStats/files/11225445/pointsND.zip)
    - [pointsNY.zip](https://github.com/USGS-WiM/StreamStats/files/11225461/pointsNY.zip)
    - [pointsNC.zip](https://github.com/USGS-WiM/StreamStats/files/11225464/pointsNC.zip)
- An example of how to create formdata object can be found in the help controller: https://github.com/USGS-WiM/StreamStats/blob/4d7c0aafda10987e85ee37e11f43676c9f604abb/src/Controllers/HelpController.ts#L107-L146
- If you google 'angularjs POST formdata' there should be more examples online as well.

**Comment by alaws-USGS** [comment]
For this issue, do you still want a message saying a batch was rejected (checkbox 3)? The instructions for submission outlined in https://github.com/USGS-WiM/StreamStats/issues/1461 say the first 250 points will be processed.

**Comment by amedenblik** [comment]
Yes, let's go ahead and keep that message in case the user missed the instruction. Slight edit suggestion:
"Batch contains more than 250 points. Only the first 250 points will be processed. Please resubmit the batch if you would like only the first 250 points to be processed."

**Comment by alaws-USGS** [comment]
I think an additional change we could add is the submit button text will change to say "Submit > 250 points". Thoughts @amedenblik and @harper-wavra?

**Comment by amedenblik** [comment]
Yeah I'm open to trying that! Along with some kind of explanatory message.

**Comment by alaws-USGS** [comment]
@amedenblik and @harper-wavra, have either of you run into this 500: 
![image](https://github.com/USGS-WiM/StreamStats/assets/108412194/4f4c0c40-e958-4eee-9590-4f189378c3f3)

It appears to be a potential issue with the API and shows with each of the files (NY, ND, NC)
. If you google the error you quite a few posts, with [this one](https://stackoverflow.com/questions/47160211/why-doesnt-tempfile-spooledtemporaryfile-implement-readable-writable-seekable) being one of the more popular.

**Comment by amedenblik** [comment]
I will look into this (https://code.usgs.gov/StreamStats/web-services-and-apis/ss-batchprocessor/-/issues/61). I haven't run into that error before when testing the services directly at https://streamstats.usgs.gov/batchprocessor/docs

**Comment by alaws-USGS** [comment]
Sounds good. I pushed my code for the formdata and double checking that the zip file is going to be formatted properly.

**Comment by alaws-USGS** [comment]
I also got the error when I tried using the services directly.

**Comment by amedenblik** [comment]
Ah ok, good to know! I may be misremembering; not sure how much we've tested those services when hosted on the server.

**Comment by amedenblik** [comment]
@alaws-USGS I believe the error has been fixed

**Comment by alaws-USGS** [comment]
@amedenblik sounds good, I'll give it a go.

**Comment by amedenblik** [comment]
Closed with #1645
